x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/rest/action/RestPutSamlServiceProviderAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.idp.saml.rest.action;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestRequest.Method;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.xpack.idp.action.PutSamlServiceProviderAction;
+import org.elasticsearch.xpack.idp.action.PutSamlServiceProviderRequest;
+import org.elasticsearch.xpack.idp.action.PutSamlServiceProviderResponse;
+
+import java.io.IOException;
+import java.util.List;
+
+public class RestPutSamlServiceProviderAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        return "idp_put_saml_sp_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(Method.PUT, "/_idp/saml/sp/{entity_id}"),
+            new Route(Method.POST, "/_idp/saml/sp/{entity_id}")
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
+        final String entityId = restRequest.param("entity_id");
```

Comment:
You either rename this to `sp_entity_id` or we rename the parameter in the get metadata API to `entity_id`, for consistency

Reply:
I can rename this, but I don't feel it's truly necessary.

The fact that this is a SP entity id is obvious from the URL, that's not true for the metadata API where there could be some confusion since it's an API that is seemingly related to the IdP, but needs to know the SP's id.

Reply:
I still think it makes sense to keep the same parameter name for the same thing in those closely related APIs